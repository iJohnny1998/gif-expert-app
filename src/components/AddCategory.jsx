import { useState } from "react"

export const AddCategory = ({ onAddCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => setInputValue(target.value);

    const onSubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim().length <= 1)
            return;

        onAddCategory(inputValue.trim());

        //Otra forma obteniendo el setCategories
        // setCategories(categories => [inputValue, ...categories])        

        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar GIFS"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}
