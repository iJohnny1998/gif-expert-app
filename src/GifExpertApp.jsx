import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball']);

    const onAddCategory = (category) => {

        if (categories.includes(category)) {
            alert(`La categor√≠a ${category} ya existe`);
            return;
        }

        return setCategories([category, ...categories]);
    }


    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory onAddCategory={onAddCategory} />

            {
                categories.map(category =>
                    <GifGrid key={category} category={category} />
                )
            }
        </>
    )
}

export default GifExpertApp
