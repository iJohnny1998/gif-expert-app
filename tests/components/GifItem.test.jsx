import { render } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Pruebas en GifItem', () => {
    const title = 'Kokun';
    const url = 'https://goku.com/kokun.jpg'

    test('Match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)

        expect(container).toMatchSnapshot();
    })

})
