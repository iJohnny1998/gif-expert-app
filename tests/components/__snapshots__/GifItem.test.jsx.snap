// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Kokun"
      src="https://goku.com/kokun.jpg"
    />
    <p>
      Kokun
    </p>
  </div>
</div>
`;
